### PR TITLE
Add semaphore.sh

### PR DESCRIPTION
**Pull request by schu**
* [x] Requires rebase after #1 was merged

### CODE REVIEW

**Comment by alban** [comment]
@schu this needs to be rebased.

**Comment by schu** [comment]
Rebased.

File: semaphore.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  rm -f ./rkt-uuid
+
+  sudo timeout --foreground --kill-after=10 5m \
+    ./rkt/rkt \
+    run --interactive \
+    --uuid-file-save=./rkt-uuid \
+    --insecure-options=image,all-run \
+    --dns=8.8.8.8 \
+    --stage1-name="kinvolk.io/aci/rkt/stage1-kvm:${rkt_version},kernelversion=${kernel_version}" \
```
Comment (alban) [inline semaphore.sh]: You could put the name of the stage1 in a variable and add a comment saying that it comes from
https://github.com/kinvolk/stage1-builder

Comment (schu) [reply semaphore.sh]: Done, thanks.

File: pkg/tracer/tracer.go [hunk 2]
```diff
@@ -48,13 +48,15 @@ func NewTracerFromFile(fileName string, tcpEventCbV4 func(TcpV4), tcpEventCbV6 f
 
 	go func() {
 		for {
+			fmt.Println("wait for ipv4 data")
```
Comment (alban) [inline pkg/tracer/tracer.go]: @schu remaining debugs?

Comment (schu) [reply pkg/tracer/tracer.go]: Oops, removed.

File: semaphore.sh [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    --environment=GOPATH=/go \
+    --environment=C_INCLUDE_PATH="${kernel_header_dir}/arch/x86/include:${kernel_header_dir}/arch/x86/include/generated" \
+    --exec=/bin/bash -- -o xtrace -c \
+      'cd /go/src/github.com/weaveworks/tcptracer-bpf/tests && \
+        mount -t tmpfs tmpfs /tmp && \
+        mount -t debugfs debugfs /sys/kernel/debug/ && \
+        make && \
+        ./run'
+
+  test_status=$(sudo ./rkt/rkt status "$(<rkt-uuid)" | awk '/app-/{split($0,a,"=")} END{print a[2]}')
```
Comment (alban) [inline semaphore.sh]: You could add a comment to say we need this because of
https://github.com/coreos/rkt/issues/2777

Comment (schu) [reply semaphore.sh]: Done as well.

**Comment by schu** [comment]
Build on Semaphore CI: https://semaphoreci.com/schu/tcptracer-bpf/branches/schu-ci/builds/8

**Comment by alban** [comment]
LGTM

**Comment by iaguis** [comment]
lgtm

**Review by iaguis — APPROVED** [review]

**Review by alban — APPROVED** [review]

**Comment by 2opremio** [comment]
Please embed https://github.com/weaveworks/build-tools in a subtree (like in Scope) and make sure shell-scripts are properly linted (the earlier the better).

File: semaphore.sh [hunk 4]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (2opremio) [inline semaphore.sh:1]: Please add a comment about what this script does

**Comment by 2opremio** [comment]
I don't know enough about rocket to comment on the functionality of this PR, so once my comments are addressed this should be good to go.

**Comment by schu** [comment]
> Please embed https://github.com/weaveworks/build-tools in a subtree (like in Scope) and make sure shell-scripts are properly linted (the earlier the better).

Done.

File: semaphore.sh (hunk 4 above)
Comment (schu) [reply semaphore.sh:1]: Done.

**Comment by 2opremio** [comment]
> Done.

Where is the linter invoked?

File: circle.yml [hunk 5]
```diff
@@ -15,6 +15,10 @@ machine:
 dependencies:
   override:
     - |
+      sudo apt-add-repository 'deb http://archive.ubuntu.com/ubuntu trusty-backports main restricted universe' &&
+      sudo apt-get update &&
+      sudo apt-get install -y shellcheck
```
Comment (2opremio) [inline circle.yml]: Please do this inside the build container instead.

File: circle.yml [hunk 6]
```diff
@@ -23,6 +27,8 @@ dependencies:
 test:
   override:
     - |
+      ./tools/shell-lint .
```
Comment (2opremio) [inline circle.yml]: Please run the full linter

File: circle.yml (hunk 5 above)
Comment (2opremio) [reply circle.yml]: And add a `lint` make target so that devs can easily invoke it.

**Comment by schu** [comment]
@2opremio all done, I believe. Note that there are 3 remaining issues:

* shfmt has a bug: https://github.com/mvdan/sh/issues/64
* build-tools itself gives a couple of warnings: https://github.com/weaveworks/build-tools/pull/71
* a go vet issue: https://github.com/weaveworks/tcptracer-bpf/pull/8

Thanks for reviewing!

File: circle.yml [hunk 7]
```diff
@@ -28,16 +28,14 @@ test:
     - |
       cd $SRCDIR && sudo docker run --rm \
                 -e GOPATH=/go \
-                -e SRCDIR=$SRCDIR2 \
                 -v $SRCDIR:$SRCDIR2 \
                 --workdir=$SRCDIR2 \
                 weaveworks/tcptracer-bpf-builder \
                 /bin/sh -c \
-                'cd tests && make && GOOS=darwin make'
+                'export PATH=$GOPATH/bin:$PATH && make lint && cd tests && make && GOOS=darwin make'
```
Comment (2opremio) [inline circle.yml]: Can you use separate lines? (it makes it easier to debug)

**Comment by schu** [comment]
CircleCI is green now.

**Review by 2opremio — APPROVED** [review]

**Comment by schu** [comment]
Thanks!

**Pull request merged**